traveler-ios-ui/TravelerKitUI/ViewControllers/CatalogItemResultViewController.swift

```diff
@@ -81,6 +85,23 @@ class CatalogItemResultViewController: UIViewController {
     @IBAction func didSelectTermsAndConditions(_ sender: Any) {
         performSegue(withIdentifier: "termsAndConditionsSegue", sender: nil)
     }
+
+    @IBAction func didToggleWishlist(_ sender: Any) {
+        guard let detail = catalogItemDetails else {
+            return
+        }
+
+        // Since isWishlisted is changed to optional now, do I show the `unidentifiedUser` error right here when isWishlisted is found to be nil or should I let the core SDK handle it?
+        if detail.isWishlisted == true {
+            Traveler.wishlistRemove([detail], result: nil, delegate: self)
+            wishlistButton.isFilled = false
+        } else {
+            Traveler.wishlistAdd([detail], delegate: self)
+            wishlistButton.isFilled = true
+        }
+
+        wishlistButton.isEnabled = false
```

Comment:
Good attention here +1

However in this particular case, if the user does tap immediately after they should see expected behaviour

We dont want to make the user wait on this call

Reply:
You didnt address this comment

Reply:
TBD